onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -24,6 +24,17 @@
 #pragma warning(disable : 26426)
 #endif
 namespace ONNX_NAMESPACE {
+
+inline int64_t HandleNegativeAxis(int64_t axis, int64_t rank) {
+  if (axis >= rank || axis < -rank) {
+    fail_shape_inference("axis ", axis,
+                         " is not in valid range [-", rank, ",", rank - 1, "]");
+  }
+
+  // Handle negative axis
+  return axis < 0 ? axis + rank : axis;
```

Comment:
So "axis + rank" still could overflow?

Reply:
axis is in [-rank, -1], then axis + rank is in [0, rank-1]